fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneDocumentFromRecordTest.java

```diff
@@ -139,6 +211,26 @@ void map() {
                         textField("k1", "v1"),
                         textField("k2", "v2"))),
                 LuceneDocumentFromRecord.getRecordFields(index, record));
+
+        Descriptors.Descriptor recordDescriptor = message.getDescriptorForType();
+        Message.Builder builder = TestRecordsTextProto.MapDocument.newBuilder();
+        LuceneIndexKeyValueToPartialRecordUtils.buildPartialRecord(index, recordDescriptor, builder, "k1", "suggestion");
+        Message partialMsg = builder.build();
+
+        Descriptors.FieldDescriptor entryField = recordDescriptor.findFieldByName("entry");
+        assertEquals(1, partialMsg.getRepeatedFieldCount(entryField));
+
+        TestRecordsTextProto.MapDocument.Entry entry = (TestRecordsTextProto.MapDocument.Entry) partialMsg.getRepeatedField(entryField, 0);
+
+        // The k1 is supposed to show up in the key field within repeated entry sub-message
+        Descriptors.FieldDescriptor keyField = TestRecordsTextProto.MapDocument.Entry.getDescriptor().findFieldByName("key");
+        assertTrue(entry.hasField(keyField));
+        assertEquals("k1", entry.getField(keyField));
+
+        // The suggestion is supposed to show up in value field within repeated entry sub-message
+        Descriptors.FieldDescriptor valueField = TestRecordsTextProto.MapDocument.Entry.getDescriptor().findFieldByName("value");
+        assertTrue(entry.hasField(valueField));
+        assertEquals("suggestion", entry.getField(valueField));
```

Comment:
```suggestion
        Descriptors.Descriptor recordDescriptor = message.getDescriptorForType();
        TestRecordsTextProto.MapDocument.Builder builder = TestRecordsTextProto.MapDocument.newBuilder();
        LuceneIndexKeyValueToPartialRecordUtils.buildPartialRecord(index, recordDescriptor, builder, "k1", "suggestion");
        TestRecordsTextProto.MapDocument partialMsg = builder.build();

        assertEquals(1, partialMsg.getEntryCount());

        TestRecordsTextProto.MapDocument.Entry entry = partialMsg.getEntry(0);

        // The k1 is supposed to show up in the key field within repeated entry sub-message
        assertEquals("k1", entry.getKey());

        // The suggestion is supposed to show up in value field within repeated entry sub-message
        assertEquals("suggestion", entry.getValue());
```

Reply:
(No promise that actually compiles, but I hope the idea is clear.) All this dynamic message stuff seems unnecessary, since already gave it a typed `Builder`.

Reply:
Fixed all tests in ```LuceneDocumentFromRecordTest``` in this way.